processor.go

```diff
@@ -96,204 +116,193 @@ func (p *Processor[AC, OC, JC]) init() {
 	}
 	// Make a map of triggers to states so we can easily reference it
 	p.stateMap = map[string]State[AC, OC, JC]{}
-	for _, s := range p.states {
-		p.stateMap[s.TriggerState] = s
-	}
-	// get a list of return state names for use
-	p.stateNames = make([]string, 0, len(p.stateMap))
-	for k := range p.stateMap {
-		p.stateNames = append(p.stateNames, k)
-	}
-
-	// For each state, we need a channel of jobs
+	p.stateStatusMap = map[string]*StatusCount{}
+	p.stateWaitingJobsMap = map[string][]Job[JC]{}
 	p.stateChan = map[string]chan Job[JC]{}
+	p.sortedStateNames = []string{}
```

Comment:
If we made a States type this could easily go in there

Reply:
fixed in v2